### PR TITLE
GH-10: Add basic checks into pre-commit-config

### LINKED ISSUE #10: pre-commit: ShellCheck
See also: https://github.com/apache/arrow-go/blob/e112ad0de5ca66133256f2db71bda4dd2d1731f5/.pre-commit-config.yaml#L46-L49

### LINKED ISSUE #11: pre-commit: shfmt
See also: https://github.com/apache/arrow-go/blob/e112ad0de5ca66133256f2db71bda4dd2d1731f5/.pre-commit-config.yaml#L50C11-L57

### LINKED ISSUE #12: pre-commit: yamlfmt
See also: https://github.com/apache/arrow-go/blob/e112ad0de5ca66133256f2db71bda4dd2d1731f5/.pre-commit-config.yaml#L58-L61

### PR DESCRIPTION
**Pull request by kiszk**
Fixes #10
Fixes #11
`Removed gh 12 now`

### CODE REVIEW

**Comment by kou** [comment]
Could you add https://github.com/apache/arrow-go/blob/main/.editorconfig for shfmt?

**Comment by kiszk** [comment]
Good catch, thank you.

File: .editconfig
Comment (kou) [inline .editconfig:1]: Could you rename this to `.editorconfig`?

Comment (kiszk) [reply .editconfig:1]: Thanks, good catch

**Review by lidavidm — COMMENTED** [review]
Do we want to revert the YAML changes since we aren't using yamlfmt?

**Comment by kiszk** [comment]
I see. I thought the reformatting of indentation made sense.
I will revert the changes in the YAML files.

**Comment by kiszk** [comment]
Reverted changes in YAML files

**Review by lidavidm — APPROVED** [review]

**Pull request merged**